core/domain/user_jobs_test.py

```diff
@@ -460,11 +479,128 @@ def test_subscribing_to_exp_subscribes_to_its_feedback_threads(self):
+            ModifiedRecentUpdatesAggregator.start_computation()
+            self.assertEqual(
+                self.count_jobs_in_taskqueue(
+                    queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+                1)
+            self.process_and_flush_pending_tasks()
+
+            recent_notifications = (
+                ModifiedRecentUpdatesAggregator.get_recent_notifications(
+                    USER_ID)[1])
+            self.assertEqual([{
+                'type': feconf.UPDATE_TYPE_COLLECTION_COMMIT,
+                'last_updated_ms': expected_last_updated_ms,
+                'activity_id': COLLECTION_ID,
+                'activity_title': 'A new title',
+                'author_id': ANOTHER_USER_ID,
+                'subject': 'Update collection',
+            }], recent_notifications)
+
+    def test_basic_computation_works_if_collection_is_deleted(self):
+        with self.swap(
+                jobs_registry, 'ALL_CONTINUOUS_COMPUTATION_MANAGERS',
+                self.ALL_CONTINUOUS_COMPUTATION_MANAGERS_FOR_TESTS):
+            COLLECTION_ID = 'cid'
+            COLLECTION_TITLE = 'Title'
+            USER_ID = 'user_id'
+
+            self.save_new_default_collection(
+                COLLECTION_ID, USER_ID, title=COLLECTION_TITLE)
+            last_updated_ms_before_deletion = (
+                self._get_most_recent_collection_snapshot_created_on_ms(
+                    COLLECTION_ID))
+            collection_services.delete_collection(USER_ID, COLLECTION_ID)
+
+            ModifiedRecentUpdatesAggregator.start_computation()
+            self.assertEqual(
+                self.count_jobs_in_taskqueue(
+                    queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+                1)
+            self.process_and_flush_pending_tasks()
+
+            recent_notifications = (
+                ModifiedRecentUpdatesAggregator.get_recent_notifications(
+                    USER_ID)[1])
+            self.assertEqual(len(recent_notifications), 1)
+            self.assertEqual(sorted(recent_notifications[0].keys()), [
+                'activity_id', 'activity_title', 'author_id',
+                'last_updated_ms', 'subject', 'type'])
+            self.assertDictContainsSubset({
+                'type': feconf.UPDATE_TYPE_COLLECTION_COMMIT,
+                'activity_id': COLLECTION_ID,
+                'activity_title': COLLECTION_TITLE,
+                'author_id': USER_ID,
+                'subject': feconf.COMMIT_MESSAGE_COLLECTION_DELETED,
+            }, recent_notifications[0])
+            self.assertLess(
+                last_updated_ms_before_deletion,
+                recent_notifications[0]['last_updated_ms'])
+
 
 class DashboardSubscriptionsOneOffJobTests(test_utils.GenericTestBase):
     """Tests for the one-off dashboard subscriptions job."""
     EXP_ID = 'exp_id'
     EXP_ID_2 = 'exp_id_2'
+    COLLECTION_ID = 'col_id'
+    COLLECTION_ID_2 = 'col_id_2'
+    COLLECTION_EXP_ID = 'col_exp_id'
```

Comment:
Took a while for me to figure out what this means. Maybe call it EXP_ID_FOR_COLLECTION_1 and s/COLLECTION_ID/COLLECTION_ID_1, s/EXP_ID/EXP_ID_1